source/source_integration_test.go

```diff
@@ -644,8 +645,21 @@ func createTestBucket(t *testing.T, client *s3.Client, bucket string) {
 		t.Logf("created test bucket %q in %v", bucket, time.Since(start))
 	}()
 
+	// this allows the integration tests to run in buckets that are not in us-east-1
+	region := os.Getenv("AWS_REGION")
+	var createConfig *types.CreateBucketConfiguration
+	for _, value := range types.BucketLocationConstraint.Values("") {
```

Comment:
What's this part about?

Reply:
My aws account did put my s3 into frankfurt for some reason. When I tried to run the integration tests with my account credentials I got a bucket creation error, something about requiring a location constraint. This did fix it.

Reply:
Just added better comment! At the end was more about the test s3 client using an aws region different than the default one, `us-east-1`. Should be a bit easier to understand now.